code/lib/preview-api/src/modules/store/csf/prepareStory.ts

```diff
@@ -98,26 +99,57 @@ export function prepareStory<TRenderer extends Renderer>(
     storyAnnotations?.render ||
     componentAnnotations.render ||
     projectAnnotations.render;
-  if (!render) throw new Error(`No render function available for storyId '${id}'`);
 
   const decoratedStoryFn = applyHooks<TRenderer>(applyDecorators)(undecoratedStoryFn, decorators);
   const unboundStoryFn = (context: StoryContext<TRenderer>) => decoratedStoryFn(context);
 
   const playFunction = storyAnnotations?.play ?? componentAnnotations?.play;
 
+  const mountUsed = mountDestructured(playFunction);
+
+  if (!render && !mountUsed) {
+    throw new Error(`No render function available for storyId '${id}'`);
+  }
+
+  let { tags } = partialAnnotations;
+
+  if (mountUsed) {
+    // Play is not supported in docs yet, and when mount is used, the mounting is happening in play itself.
+    tags = tags.filter((tag) => tag !== 'autodocs');
```

Comment:
What does this do exactly?

Reply:
I tried to improve the message.

Reply:
// Don't show stories where mount is used in docs.
    // As the play function is not running in docs, and when mount is used, the mounting is happening in play itself.

Reply:
Maybe:

```
// Hide stories that mount inside the `play` function from docs
// Docs doesn't currently run the `play` function, so such stories will never be mounted.
```

This does seem like the sort of thing that will confuse users though ("why has my story disappeared from docs?"). I wonder if there's some way we could make it more obvious for folks.

Also I wonder if we should alter the tag in the index rather than dynamically here. WDYT @shilman?

Reply:
Working on a new PR that fixes this